app/org/sagebionetworks/bridge/hibernate/HibernateAccountDao.java

```diff
@@ -313,22 +293,25 @@ public void createAccountForMigration(Study study, Account account, String id) {
         try {
             hibernateHelper.create(hibernateAccount);
         } catch (ConcurrentModificationException ex) {
-            // Account can conflict because studyId + email or studyId + phone has been used for an 
+            // Account can conflict because studyId + email|phone|externalId|healthCode have been used for an 
             // existing account. 
             AccountId accountId = null;
             if (hibernateAccount.getEmail() != null) {
                 accountId = AccountId.forEmail(study.getIdentifier(), account.getEmail());
             } else if (hibernateAccount.getPhone() != null) {
                 accountId = AccountId.forPhone(study.getIdentifier(), account.getPhone());
+            } else if (hibernateAccount.getExternalId() != null) {
+                accountId = AccountId.forExternalId(study.getIdentifier(), account.getExternalId());
```

Comment:
Should we add a unit test to HibernateAccountDaoTest in the style of createAccountAlreadyExists() and createAccountAlreadyExistsForPhoneAccount()?

Reply:
Added createAccountAlreadyExistsForExternalIdAccount() where you're creating an account with only an external ID. It's used to retrieve the existing account.